src/rules/selector-pseudo-class-whitelist/__tests__/index.js

```diff
@@ -0,0 +1,99 @@
+import {
+  messages,
+  ruleName,
+} from ".."
+import rules from "../../../rules"
+import { testRule } from "../../../testUtils"
+
+const rule = rules[ruleName]
+
+testRule(rule, {
+  ruleName,
+  config: [[
+    "hover",
+    "nth-child",
+    "root",
+    "placeholder",
+  ]],
+  skipBasicChecks: true,
+
+  accept: [ {
+    code: "a {}",
+  }, {
+    code: "a:hover {}",
+  }, {
+    code: "a:nth-child(5) {}",
+  }, {
+    code: ":root {}",
+  }, {
+    code: "a:HOVER {}",
+  }, {
+    code: "a:hover, a:nth-child(5) {}",
+  }, {
+    code: "a::before {}",
+  }, {
+    code: "a:nth-child(5)::before {}",
+  }, {
+    code: "a:-moz-placeholder {}",
+  } ],
+
+  reject: [ {
+    code: "a:focus {}",
+    message: messages.rejected("focus"),
+    line: 1,
+    column: 2,
+  }, {
+    code: "div:nth-LAST-child {}",
+    message: messages.rejected("nth-LAST-child"),
+    line: 1,
+    column: 4,
+  }, {
+    code: "a,\n:global {}",
+    message: messages.rejected("global"),
+    line: 2,
+    column: 1,
+  }, {
+    code: "input:-ms-input-placeholder {}",
```

Comment:
Add test `input:-MS-INPUT-PLACEHOLDER {}` to tests case insensitivity for vendor properties